src/App.jsx

```diff
@@ -1,21 +1,40 @@
-
-import './App.css'
-import { Client } from 'appwrite'; 
+import React, { useState, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import "./App.css";
+import { login, logout } from "./store/authSlice";
+import authService from "./appwrite/auth";
+import { Header, Footer } from "../src/components/index";
 
 function App() {
-console.log(import.meta.env.VITE_APPWRITE_URL)
-const client = new Client();
-
-client
-    .setEndpoint('https://cloud.appwrite.io/v1')
-    .setProject('65e5c4c6b5cfc4dda432');
+  const [loading, setLoading] = useState(true);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    authService
```

Comment:
1. keep this component clean unless much needed. Shift the API call and render to the child component and use that component here
2. Try to give single responsibility to component. For example, in one component only API call will happen and then API data will flow down through props to the stateless comp
3. Try to have container/presentational architc ( although its not mandatory )